cluster-autoscaler/core/scale_up.go

```diff
@@ -35,9 +35,129 @@ import (
 	"k8s.io/autoscaler/cluster-autoscaler/utils/nodegroupset"
 	schedulercache "k8s.io/kubernetes/pkg/scheduler/cache"
 
+	"fmt"
 	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"math"
 )
 
+type scaleUpResourcesLimits map[string]int64
+type scaleUpResourcesDelta map[string]int64
+
+// used as a value in scaleUpResourcesLimits if actual limit could not be obtained due to errors talking to cloud provider
+const scaleUpLimitUnknown = math.MaxInt64
+
+func computeScaleUpResourcesLeftLimits(nodeGroups []cloudprovider.NodeGroup, nodeInfos map[string]*schedulercache.NodeInfo, resourceLimiter *cloudprovider.ResourceLimiter) scaleUpResourcesLimits {
+	totalCores, totalMem, errCoresMem := calculateScaleUpCoresMemoryTotal(nodeGroups, nodeInfos)
+
+	resultScaleUpLimits := make(scaleUpResourcesLimits)
+	for _, resource := range resourceLimiter.GetResources() {
+		max := resourceLimiter.GetMax(resource)
+
+		// we put only actual limits into final map. No entry means no limit.
+		if max > 0 {
+			switch {
+			case resource == cloudprovider.ResourceNameCores:
+				if errCoresMem != nil {
+					resultScaleUpLimits[resource] = scaleUpLimitUnknown
```

Comment:
Setting unknown instead of immediately failing whole scale-up with error only makes sense if we consider nodes without CPU? I know it's technically possible to have a Node object without CPU resource, but I'm not sure how practical it is to consider that. Also IIRC in scale-down you don't allow unknown CPU.

Same for memory.